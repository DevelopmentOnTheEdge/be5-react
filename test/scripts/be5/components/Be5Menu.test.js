import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, mount, render} from 'enzyme';
import {TestProvider}      from "../testUtils";
import bus            from '../../../../src/scripts/be5/core/bus';
import ApplicationWithBe5Menu            from '../../../../src/scripts/be5/components/test/ApplicationWithBe5Menu';
import Be5Menu            from '../../../../src/scripts/be5/components/be5Menu/Be5Menu';
import Be5MenuItem        from '../../../../src/scripts/be5/components/be5Menu/Be5MenuItem';
import '../mockBe5Request'


test('Be5Menu', () => {

  const component = renderer.create(
    <TestProvider>
      <Be5Menu show={true}/>
    </TestProvider>
  );

  bus.fire("RefreshAll");

  expect(component.toJSON()).toMatchSnapshot();


  // changeDocument('MainDocument', { component: StaticPage, value: "test" });
  // expect(component.toJSON()).toMatchSnapshot();

});

test('Be5MenuItem', () => {

  const component = renderer.create(
    <TestProvider>
     <Be5MenuItem entity="testtable" view="Test 1D"/>
    </TestProvider>
  );

  bus.fire("RefreshAll");

  expect(component.toJSON()).toMatchSnapshot();


  // changeDocument('MainDocument', { component: StaticPage, value: "test" });
  // expect(component.toJSON()).toMatchSnapshot();

});

test('ApplicationWithBe5Menu', () => {

  const component = renderer.create(
    <TestProvider>
      <ApplicationWithBe5Menu />
    </TestProvider>
  );

  bus.fire("RefreshAll");

  expect(component.toJSON()).toMatchSnapshot();


  // changeDocument('MainDocument', { component: StaticPage, value: "test" });
  // expect(component.toJSON()).toMatchSnapshot();

});
