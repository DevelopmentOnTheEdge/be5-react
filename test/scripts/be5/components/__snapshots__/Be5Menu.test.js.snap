// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationWithBe5Menu 1`] = `
<div>
  <div>
    <div
      className="s-alert-wrapper"
    />
  </div>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark"
  >
    <div
      className="container"
    >
      <button
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="dropdown nav-item"
            onKeyDown={[Function]}
          >
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle nav-link"
              href="#"
              onClick={[Function]}
            >
              Test table
            </a>
            <div
              aria-hidden={true}
              className="dropdown-menu"
              role="menu"
              tabIndex="-1"
            >
              <a
                className="dropdown-item"
                href="#!table/testtable/Test 1D"
                onClick={[Function]}
                tabIndex="0"
                type={undefined}
              >
                Test 1D
              </a>
              <a
                className="dropdown-item"
                href="#!table/testtable/Test 1D unknown"
                onClick={[Function]}
                tabIndex="0"
                type={undefined}
              >
                Test 1D unknown
              </a>
            </div>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="#!table/users/All records"
              onClick={[Function]}
            >
              Пользователи
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="#!logout"
              onClick={[Function]}
            >
              Выход
            </a>
          </li>
        </ul>
        <form
          className="form-inline ml-auto"
        >
          <a
            className="btn btn-secondary"
            href="#!logout"
            role="button"
          >
            Logout
          </a>
        </form>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div
      className="row"
    />
  </div>
</div>
`;

exports[`Be5Menu 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <button
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      style={Object {}}
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="dropdown nav-item"
          onKeyDown={[Function]}
        >
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle nav-link"
            href="#"
            onClick={[Function]}
          >
            Test table
          </a>
          <div
            aria-hidden={true}
            className="dropdown-menu"
            role="menu"
            tabIndex="-1"
          >
            <a
              className="dropdown-item"
              href="#!table/testtable/Test 1D"
              onClick={[Function]}
              tabIndex="0"
              type={undefined}
            >
              Test 1D
            </a>
            <a
              className="dropdown-item"
              href="#!table/testtable/Test 1D unknown"
              onClick={[Function]}
              tabIndex="0"
              type={undefined}
            >
              Test 1D unknown
            </a>
          </div>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="#!table/users/All records"
            onClick={[Function]}
          >
            Пользователи
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="#!logout"
            onClick={[Function]}
          >
            Выход
          </a>
        </li>
      </ul>
      <form
        className="form-inline ml-auto"
      >
        <a
          className="btn btn-secondary"
          href="#!logout"
          role="button"
        >
          Logout
        </a>
      </form>
    </div>
  </div>
</nav>
`;

exports[`Be5MenuItem 1`] = `
<a
  className="menu-item"
  href="#!table/testtable/Test 1D"
  target=""
>
  Test 1D
</a>
`;
